drape_frontend/my_position_controller.cpp

```diff
@@ -670,10 +672,18 @@ void MyPositionController::StopLocationFollow()
 
 void MyPositionController::OnEnterForeground(double backgroundTime)
 {
-  if (backgroundTime >= kMaxTimeInBackgroundSec && m_mode == location::NotFollow)
+  if (backgroundTime >= kMaxTimeInBackgroundSec)
   {
-    ChangeMode(m_isInRouting ? location::FollowAndRotate : location::Follow);
-    UpdateViewport(kDoNotChangeZoom);
+      if (m_mode == location::NotFollow)
+      {
+          ChangeMode(m_isInRouting ? location::FollowAndRotate : location::Follow);
+          UpdateViewport(kDoNotChangeZoom);
+      }
+      else if (m_mode == location::NotFollowNoPosition)
+      {
+          ChangeMode(location::PendingPosition);
+          UpdateViewport(kDoNotChangeZoom);
```

Comment:
Should the viewport be updated when we switch from no position to pending position? What does UpdateViewport do?

Reply:
Good point, thanks! I checked it out - UpdateViewport does nothing when the position mode is Pending - an unnecessary call.